table/block_based/filter_policy.cc

```diff
@@ -33,28 +35,28 @@
 #include "util/ribbon_config.h"
 #include "util/ribbon_impl.h"
 #include "util/string_util.h"
+#include "utilities/memory_allocators.h"
 
 namespace ROCKSDB_NAMESPACE {
 
 namespace {
 
+DefaultMemoryAllocator kDefaultAllocator;
+
 // Metadata trailer size for built-in filters. (This is separate from
 // block-based table block trailer.)
 //
 // Originally this was 1 byte for num_probes and 4 bytes for number of
 // cache lines in the Bloom filter, but now the first trailer byte is
 // usually an implementation marker and remaining 4 bytes have various
 // meanings.
-static constexpr uint32_t kMetadataLen = 5;
+constexpr uint32_t kMetadataLen = 5;
 
-Slice FinishAlwaysFalse(std::unique_ptr<const char[]>* /*buf*/) {
-  // Missing metadata, treated as zero entries
-  return Slice(nullptr, 0);
-}
+// Missing metadata, treated as zero entries
+const Slice kAlwaysFalseFilter = Slice();
 
-Slice FinishAlwaysTrue(std::unique_ptr<const char[]>* /*buf*/) {
-  return Slice("\0\0\0\0\0\0", 6);
-}
+// Treated as legit filter with zero probes
+const Slice kAlwaysTrueFilter("\0\0\0\0\0\0", 6);
```

Comment:
I might be missing something but it seems to me that Slice::remove_prefix/remove_suffix() can modify this static object by shortening it, affecting other reference to this object.